[B + C] Fixed HumanEntity#openInventory not being up to date, Fixes BUKKIT-3690

thefishlive opened the pull request:
#### Description:

CraftEntityHuman#openInventory(Inventory) is missing the code to deal with Anvil/Beacon/Trade/Horse inventory types and bukkit is missing a Horse inventory type, and if a unknown type is passed to it the method will fail silently.
#### Justification and breakdown

Any method should not fail silently if an error has occurred, and the new inventory types should be able to be opened via these methods. This PR adds the new inventory types to the switch statement to handle opening of them correctly, it also adds a default case at the end so any unknown inventory types will though a IllegalArgumentException. Finally this PR adds the new inventory types to the CraftContainer so that it knows how to handle opening them correctly.
#### Testing Results and Materials:
##### TestPlugin

(Plugin allows for opening of inventories by using '/&lt;inventory type&gt;')
source - https://github.com/thefishlive/Bukkit-test/tree/inventory
binary  - https://www.dropbox.com/s/p7hibwjqb2wqo4r/bukkit-test-0.1-SNAPSHOT.jar
#### Relevant PR(s):

CB-1175 - https://github.com/Bukkit/CraftBukkit/pull/1175
Ticket: BUKKIT-3690- https://bukkit.atlassian.net/browse/BUKKIT-3690

Unh0lyTigg commented on src/main/java/org/bukkit/entity/HumanEntity.java:
```diff
@@ -52,32 +52,58 @@
+     *
+     * @param location The location to attach it to. If null, the player's
+     *            location is used.
+     * @param force If false, and there is no enchanting table at the
```
Enchanting table? For opening an anvil inventory view? You should probably fix this.

thefishlive replied:
dam it, will do that now

thefishlive committed:
Fixed HumanEntity#openInventory not being up to date, Fixes BUKKIT-3690
Now supports all the current minecraft inventories and will throw an exception if a plugin attempts to open a inventory that is not supported instead of failing silently like it did before.

thefishlive commented:
Updated to include 1.6 horse inventory

turt2live commented:
From: https://github.com/Bukkit/Bukkit/pull/853#issuecomment-16759378

> If you want to fix javadoc, I encourage you to submit patches to https://github.com/Bukkit/Bukkit-JavaDoc. As for new javadoc, it needs to meet standards. 78 characters is the rule we've been using for fixing existing and adding new javadoc.

What this means is that all the existing JavaDocs that you formatted should be reverted as it clutters the diff.

turt2live commented:
Overall: Could this PR be updated to include all the missing inventory types? 

And the title should not be "Improved" - This gives the impression that you made the code faster/more efficient.

thefishlive commented:
What other types?, If you mean Crafting/Creative I do not know entirly the reason why they were excluded before so I just left them.
And yes, I will change the PR title/commit tomorrow.

turt2live commented:
Crafting and Creative inventories, if possible, should be added :) - What I meant

thefishlive commented:
Is impossible, there is no handle for opening them on the client side as far as i can see.

riking commented:
@turt2live As far as I know, it should be impossible for the server to force the current client to open the E inventory, in both survival and creative.

<hr>

Which, as I am a doofus, is exactly what @thefishlive just said. ヽ( ｡ヮﾟ)ノ

turt2live commented:
@riking Thanks for copy/pasting and rewording :package: 

(as you pointed out)

turt2live commented on src/main/java/org/bukkit/entity/HumanEntity.java:
```diff
@@ -59,6 +59,8 @@
      *
      * @param inventory The inventory to open
      * @return The newly opened inventory view
+     * @throws IllegalArgumentException if the specified inventory cannot be
```
This JavaDoc does not belong here. It should be implemented in Bukkit-JavaDoc, not here

turt2live commented on src/main/java/org/bukkit/entity/HumanEntity.java:
```diff
@@ -83,6 +85,19 @@
+     * @param force If false, and there is no anvil at the location, no
+     *            inventory will be opened and null will be returned.
+     * @return The newly opened inventory view, or null if it could not be
+     *         opened.
```
Should be 4 spaces in

turt2live commented on src/main/java/org/bukkit/entity/HumanEntity.java:
```diff
@@ -83,6 +85,19 @@
+     * @param location The location to attach it to. If null, the player's
+     *            location is used.
+     * @param force If false, and there is no anvil at the location, no
+     *            inventory will be opened and null will be returned.
```
Should be 4 spaces in

turt2live commented on src/main/java/org/bukkit/entity/HumanEntity.java:
```diff
@@ -83,6 +85,19 @@
+     * the bottom.
+     *
+     * @param location The location to attach it to. If null, the player's
+     *            location is used.
```
Should be 4 spaces in

thefishlive committed:
Fixed javadoc issues
